src/TestInfrastructure/Framework/ConfigurableConfigurationProvider.cs

```diff
@@ -57,6 +62,8 @@ public bool WriteConfiguration(BindingConfiguration configuration)
         public bool WriteResultToReturn { get; set; }
         public BindingConfiguration SavedConfiguration { get; set; }
 
+        public int DeleteCallCount { get; set; }
```

Comment:
I would have the setter `private` because it doesn't really make sense to allow testing code to change its value.